src/Events/Extensions/AppCloudEventExtensions.cs

```diff
@@ -26,12 +30,43 @@ public static CloudEvent CreateEvent(this AppCloudEventRequestModel appEvent)
                 Type = appEvent.Type
             };
 
+            var sourceProperties = GetPropertiesFromAppSource(appEvent.Source);
+
+            cloudEvent.SetAttributeFromString("resource", $"{AuthorizationConstants.AppResourcePrefix}{sourceProperties.Org}.{sourceProperties.App}");
+            cloudEvent.SetAttributeFromString("resourceinstance", $"{sourceProperties.InstanceOwnerPartyId}/{sourceProperties.InstanceGuid}");
+
             if (!string.IsNullOrEmpty(appEvent.AlternativeSubject))
             {
                 cloudEvent.SetAttributeFromString("alternativesubject", appEvent.AlternativeSubject);
             }
 
             return cloudEvent;
         }
+
+        /// <summary>
+        /// Decomposes the source of a cloud event into known parts.
+        /// </summary>
+        /// <param name="source">The uri source of the cloud event</param>
+        /// <returns>A tuple containing org, app, intanceOwnerPartyId and instanceGuid</returns>
+        /// <remarks>Domain should always be validated as a match to current environment before using this method.</remarks>
+        public static (string Org, string App, string InstanceOwnerPartyId, string InstanceGuid) GetPropertiesFromAppSource(Uri source)
```

Comment:
I assume the need for this method is temporary? I don't like where we added it, but I can close my eyes for things that are actually temporary. Will we be able to remove it again when all apps are producing events with resource and resourceinstance properties?

Reply:
I don't think this will be removed any time soon as authorization of app events will still require these parameters. I think the agreement with team authorization was that we would provide these attributed in the decision request rather than them having to pick them out on their end.

Reply:
Yes we should be able to remove it once apps start to produce these properties themselves. We will still need some decomposition of the resource and resourceinstance to correcly populate the decision request with attributes, but that would then be limited to the xacml mapper.

Reply:
do we have a good way of making sure that org and app names don't contain delimiters that can trip us up when doing this kind of decomposition? I'm guessing that both "." and ":" could cause problems.